bot/cogs/moderation/management.py

```diff
@@ -49,8 +49,8 @@ async def infraction_group(self, ctx: Context) -> None:
     async def infraction_edit(
         self,
         ctx: Context,
-        infraction_id: t.Union[int, allowed_strings("l", "last", "recent")],
-        duration: t.Union[Expiry, allowed_strings("p", "permanent"), None],
+        infraction_id: t.Union[int, allowed_strings("l", "last", "recent")],  # noqa: F821
+        duration: t.Union[Expiry, allowed_strings("p", "permanent"), None],   # noqa: F821
```

Comment:
I'm probably missing something here, but why did you add these? I'm a bit surprised that this error code would pop-up here (unless PyFlakes does not understand `allowed_strings` with arguments) and I can't reproduce it locally with our linting set-up and these noqa's removed. 

```suggestion
        infraction_id: t.Union[int, allowed_strings("l", "last", "recent")],
        duration: t.Union[Expiry, allowed_strings("p", "permanent"), None],
```

Reply:
It's a new `flake8 3.8` error, that's probably why you can't reproduce it. I agree that it's basically a bug, but it does fail the linter at 3.8.

Reply:
Also, I know this isn't strictly relevant to this PR, it's here out of pure laziness because I didn't wanna downgrade my flake8 and pre-commit setup. but if you mind, I could.

Reply:
We're going to merge a PR that includes a relock soon-ish, right?

It would make more sense to document this in that PR so it's clear why this was changed, but it feels a bit like creating a lot of work (removing it here, adding it back in another PR).

Do you know which PR is merging that version upgrade? It'll be good to keep an eye on if it ran a full lint over the entire code or just the files changed. If the latter, it may create confusing linting errors for future contribs, as linting errors suddenly pop up in files they've edited but not in the parts they actually changed.

Reply:
The full relint is handled in https://github.com/python-discord/bot/pull/947, and yes, it is a full relint.